docs/authentication/jwt-auth.html

```diff
@@ -16,13 +16,407 @@
+        <p>Inside the app extension, we create an RSA `JWTVerifier` by providing the public key:</p>
+        <pre><code class="language-swift">static let jwtVerifier = JWTVerifier.rs256(publicKey: Data(publicKey.utf8))</code></pre>
+      </div>
+    </section>
+    <div class="underline"></div>
+
+    <h2 class="heading-2"><span class="blue-text">Step 3:</span> Define a UserCredentials Model</h2>
+    <p class="block-text">For the initial authentication, the user will have to provide their username and password.
+         This could be done through basic authentication, the Authorization header or in the body of a POST request.
+         We will go with the body of a Post request and use a model to represent the user credentials.</p>
+         <div class="info">
+           <p>Passwords and signed JWTs must be kept private and should always be exchanged over a secure layer like HTTPS.</p>
+         </div>
+        <p>Create a UserCredentials.swift file in `Sources` > `Application` > `Models`.</p>
+        <p>Inside this file we define our UserCredentials model:</p>
+        <pre><code class="language-swift">struct UserCredentials: Codable {
+    let username: String
+    let password: String
+}</code></pre>
+    <div class="underline"></div>
+
+    <h2 class="heading-2"><span class="blue-text">Step 4:</span> Authenticate the User</h2>
+    <p class="block-text">We need to read the users credentials in our POST route so they can be authenticated.</p>
+    <p>Inside the post route add:</p>
+    <pre><code class="language-swift">let credentials = try request.read(as: UserCredentials.self)
+// Users credentials are authenticated</code></pre>
+    <p>At this stage, you would normally hash the password and verify it against a database.
+        However, for simplicity, we are going to assume the user successfully logged in.</p>
+    <div class="underline"></div>
+    <h2 class="heading-2"><span class="blue-text">Step 5:</span> Create the signed JWT</h2>
+    <p class="block-text">A JWT contains claims about the user that we want include in subsequent request.
+         You can specify any information as a claim, however there are "Registered Claims" which have a pre-defined meaning:</p>
+         <ul>
+           <li>iss: The issuer of the token </li>
+           <li>sub: The subject of the token</li>
+           <li>aud: The audience of the token</li>
+           <li>exp: The expiration time which MUST be after the current date/time.</li>
+           <li>nbf: Defines the time before which the JWT MUST NOT be accepted for processing</li>
+           <li>iat: The time the JWT was issued. Can be used to determine the age of the JWT</li>
+           <li>jti: Unique identifier for the JWT. Can be used to prevent the JWT from being replayed.</li>
+         </ul>
+    <p>SwiftJWT comes with a struct representing these Registered Claims which we will use for our example.</p>
+    <p>Beneath the users credentials, initialize the users claims:</p>
```

Comment:
Inside the POST route, beneath the code where we authenticated the user, initialize the user's claims: